nethttp/client.go

```diff
@@ -159,7 +161,15 @@ func (h *Tracer) clientTrace() *httptrace.ClientTrace {
 }
 
 func (h *Tracer) getConn(hostPort string) {
-	ext.HTTPUrl.Set(h.sp, hostPort)
+	host, portString, err := net.SplitHostPort(hostPort)
+	if err != nil {
+		ext.PeerHostname.Set(h.sp, hostPort)
+	} else {
+		ext.PeerHostname.Set(h.sp, host)
+		if port, err := strconv.Atoi(portString); err != nil {
+			ext.PeerPort.Set(h.sp, uint16(port))
+		}
+	}
```

Comment:
I don't see a way to get the actual peer hostname from the request context, only via the `GotConn` callback.